docs/tutorials/4. Developer notes.md

```diff
@@ -0,0 +1,18 @@
+
+# Developer notes
```

Comment:
Is this section, mean to be empty? It's good to add some minimal information here.

Reply:
I did leave it empty, mainly because I feel like most of the recommendations are already interspersed in the main tutorials (1, 2, 3). As I said above, perhaps they make more sense in the context where they're necessary. If you think they make more sense grouped in a "Developer Notes" tutorial, I can move all of them, no prob.